libquickevent/libquickeventcore/src/exporters/stageresultscsvexporter.cpp

```diff
@@ -0,0 +1,167 @@
+#include "stageresultscsvexporter.h"
+
+#include "../og/timems.h"
+
+#include <qf/core/sql/query.h>
+#include <qf/core/sql/querybuilder.h>
+#include <qf/core/log.h>
+
+#include <QDir>
+
+namespace quickevent {
+namespace core {
+namespace exporters {
+
+StageResultsCsvExporter::StageResultsCsvExporter(QObject *parent)
+	: Super(parent)
+{
+}
+
+void StageResultsCsvExporter::generateCsvMulti()
+{
+	prepareExport();
+	exportClasses(false);
+}
+
+void StageResultsCsvExporter::generateCsvSingle()
+{
+	prepareExport();
+	exportClasses(true);
+}
+
+void StageResultsCsvExporter::exportClasses(bool single_file)
+{
+	QString where;
+	if(!classesLike().isEmpty())
+		where += "name LIKE '" + classesLike() + "'";
+	if(!classesNotLike().isEmpty()) {
+		if(!where.isEmpty())
+			where += " AND ";
+		where += "name NOT LIKE '" + classesNotLike() + "'";
+	}
+	QString qs = "SELECT id, name FROM classes";
+	if(!where.isEmpty())
+		qs += " WHERE " + where;
+	qs += " ORDER BY name";
+	qfDebug() << "loading clases:" << qs;
+
+	QString sub_dir = QString("E%1").arg(currentStage());
+	QDir csv_dir(outDir());
+	csv_dir.mkpath(sub_dir);
+
+	if (single_file) {
+		QFile fCSV(csv_dir.absolutePath() + '/' + sub_dir + '/' + "results.csv");
+		qfInfo() << "Generating:" << fCSV.fileName();
+		if (!fCSV.open(QFile::WriteOnly | QFile::Text))
```

Comment:
Please do not use TEXT open mode, we will get different files on different OS this way. https://doc.qt.io/qt-6/qiodevicebase.html#OpenModeFlag-enum

Reply:
Ok. I didn't know it